docs/guides/src/main/server/configuration.adoc

```diff
@@ -4,202 +4,237 @@
 
-By default, running the `start` command is going to automatically run the `build` command prior to starting the server. While this is a handy way to start the server with any available option,
-including **build options** that otherwise are only available to the `build` command, it introduces an additional overhead in the server startup time. For most environments, this approach is not optimal.
+You can also specify an explicit configuration file location using the `[-cf|--config-file]` option by invoking the following command:
 
-== Configuration commands and stages
+<@kc.start rootParameters="--config-file=/path/to/myconfig.conf"/>
 
-The server options are narrowed to a specific command or configuration stage. The goal is to perform a series of optimizations in a specific order to achieve optimal startup and runtime performance. This configuration occurs in two stages:
+=== Using the command-line help
+Keycloak is packed with a CLI that helps you to configure Keycloak. To find out about the available configuration, invoke the following command:
 
-First stage:: Configuration performed before starting the server in order to build an optimized server image for use at runtime
-Second stage:: Configuration performed as part of starting the server
+<@kc.start parameters="--help"/>
 
-=== First stage: Build an optimized Keycloak image
+Alternatively, you can find all server options at the <@links.server id="all-config"/> guide.
 
-The first stage involves running the `build` command and setting any **build option** available from this command to configure the server.
+=== Using raw Quarkus properties
+In most cases, the available configuration options should suffice to configure the server.
+However, you might need to use properties directly from the underlying Quarkus framework to enable a specific behavior or capability that is missing in the keycloak configuration.
 
-The configuration options at <@links.server id="all-config"/> include options that are marked with a tool icon. This icon indicates they are build options. Build options take effect only when you apply them to the `build` command.
+If possible, avoid using properties directly from Quarkus. These are considered unsupported by Keycloak. If your need is essential, consider opening an https://github.com/keycloak/keycloak/issues/new?assignees=&labels=kind%2Fenhancement%2Cstatus%2Ftriage&template=enhancement.yml[enhancement request] first and help us
+to improve Keycloak's configuration to fit your needs.
 
-You can also check which options require a build by looking at the `help` message of the `build` command:
+If that's not possible, you can configure the server using raw Quarkus properties. Perform the following steps:
 
-<@kc.build parameters="--help"/>
+. Create a `quarkus.properties` file in the `conf` directory and define any property you need.
```

Comment:
I find it a bit awkward that we say "perform the following step_s_" yet there's only a single step and it's even numbered. :) Feels a bit "robotic". :D But not a blocking issue.

Reply:
changed, you are right